stac_generator/generator.py

```diff
@@ -1,31 +1,34 @@
-"""This module encapsulates the logic for generating STAC for a given metadata standard."""
-from abc import ABC, abstractmethod
+"""This module encapsulates the logic for generating Stac for a given metadata standard."""
+
 import os
-from typing import List, Optional
+from abc import ABC, abstractmethod
+from pathlib import Path
 
 import pystac
+import requests
+
+__all__ = ("StacGenerator",)
 
 
 class StacGenerator(ABC):
-    """STAC generator base class."""
+    """Stac generator base class."""
 
-    def __init__(self, data_type, data_file, location_file) -> None:
+    def __init__(self, data_type, data_file, location_file, **kwargs) -> None:  # type: ignore[no-untyped-def]
         self.base_url = os.environ.get("STAC_API_URL", None)
         if self.base_url is None:
-            raise ValueError("Environment variable STAC_API_URL must be defined.")
+            raise ValueError("Environment variable Stac_API_URL must be defined.")
         self.data_type = data_type
         self.data_file = data_file
         self.location_file = location_file
         self.standard_file = f"./standards/{self.data_type}_standard.csv"
-        self.items: List[pystac.Item] = []
-        self.catalog: Optional[pystac.Catalog] = None
-        self.collection: Optional[pystac.Collection] = None
+        self.items: list[pystac.Item] = []
```

Comment:
I guess I need to remember that the typing module is deprecated, for some reason this is hard for me to get used to.